lib/inputstreamhelper.py

```diff
@@ -612,38 +613,39 @@ def _install_widevine_arm(self):
             self._url = arm_device['url']
             downloaded = self._http_download(message=LANGUAGE(30022))  # Downloading the recovery image
             if downloaded:
-                xbmcgui.Dialog().ok(LANGUAGE(30023), LANGUAGE(30024))
-                busy_dialog = xbmcgui.DialogBusy()
-                busy_dialog.create()
+                progress_dialog = xbmcgui.DialogProgress()
+                progress_dialog.create(heading=LANGUAGE(30043), line1=LANGUAGE(30044))  # Extracting Widevine CDM
                 bin_filename = self._url.split('/')[-1].replace('.zip', '')
                 bin_path = os.path.join(self._temp_path(), bin_filename)
 
+                progress_dialog.update(5, line1=LANGUAGE(30045), line2=LANGUAGE(30046), line3=LANGUAGE(30047))  # Uncompressing image
                 success = [
                     self._unzip(self._temp_path(), bin_filename),
                     self._check_loop(), self._set_loop_dev(),
                     self._losetup(bin_path), self._mnt_loop_dev()
                 ]
                 if all(success):
+                    progress_dialog.update(90, line1=LANGUAGE(30048))  # Extracting Widevine CDM
                     self._extract_widevine_from_img()
+                    progress_dialog.update(95, line1=LANGUAGE(30049))  # Installing Widevine CDM
                     self._install_cdm()
+                    progress_dialog.update(100, line1=LANGUAGE(30050))  # Finishing
                     self._cleanup()
                     if self._has_widevine():
                         with open(self._widevine_config_path(), 'w') as config_file:
                             config_file.write(json.dumps(devices, indent=4))
-                        xbmcgui.Dialog().notification(LANGUAGE(30037), LANGUAGE(30003))
-                        busy_dialog.close()
                         wv_check = self._check_widevine()
                         if wv_check:
-                            xbmcgui.Dialog().notification(LANGUAGE(30037), LANGUAGE(30003))
-                            busy_dialog.close()
+                            progress_dialog.update(99, line1=LANGUAGE(30003))  # Widevine installation was successful
+                        xbmcgui.Dialog().notification(LANGUAGE(30037), LANGUAGE(30003))  # Widevine installation was successful
```

Comment:
I am not so convinced about this part.

The original code send a *success* notification before doing `self._check_widevine()` and after the check (when it was successful). So basically always, and sometimes twice. I guess that was wrong.

But I don't know what we should do if `self._checkwidevine()` would return False...

Reply:
It originates from this: https://github.com/emilsvennesson/script.module.inputstreamhelper/commit/c50904cc9c85dfd413df908c0ba6c4f9643720f5